### PR TITLE
trufflehog: 2.0.97 -> 2.1.11

### PR DESCRIPTION
**Pull request by kdauler**
###### Motivation for this change
Upgrade Trufflehog to 2.1.11.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc: @kalbasit

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -25961,7 +25961,9 @@ in
 
   tusk = callPackage ../applications/office/tusk { };
 
-  trufflehog = callPackage ../tools/security/trufflehog { };
+  trufflehog = callPackage ../tools/security/trufflehog { 
+    pythonPackages = python3Packages;
```
Comment (Mic92) [inline pkgs/top-level/all-packages.nix:25965]: The nix expression is easier to read, if you just use python3Packages within the package.

**Review by bhipple — CHANGES_REQUESTED** [review]
LGTM, but let's change the internal reference to `python3Packages` explicitly, and squash up the commits.

**Comment by kdauler** [comment]
Thanks for your patience @Mic92 and @bhipple

**Comment by bhipple** [comment]
Looks like the pkg got mangled in the squash fixups!

**Comment by kdauler** [comment]
Ah yea, and mangled again

**Comment by bhipple** [comment]
LGTM now, would you mind squashing your commits down?

**Pull request merged**